atom/browser/native_window.cc

```diff
@@ -805,6 +810,25 @@ void NativeWindow::ScheduleUnresponsiveEvent(int ms) {
       base::TimeDelta::FromMilliseconds(ms));
 }
 
+void NativeWindow::SetHtmlApiFullscreen(bool enter_fullscreen) {
+  // Window is already in fullscreen mode, save the state.
+  if (enter_fullscreen && IsFullscreen()) {
+    forced_fullscreen_ = true;
+    html_fullscreen_ = true;
+    return;
+  }
+
+  // Exit html fullscreen state but not window's fullscreen mode.
+  if (!enter_fullscreen && forced_fullscreen_) {
```

Comment:
If the browser window is set with fullscreen:true or kiosk:true we only want to exit the html full screen context and not the window.